src/Http/Offer/CurrentUserHasPermissionRequestHandler.php

```diff
@@ -36,6 +36,6 @@ public function handle(ServerRequestInterface $request): ResponseInterface
 
         $hasPermission = $this->userPermissionChecker->hasPermission($offerId, $this->currentUserId);
 
-        return new UncacheableJsonResponse(['hasPermission' => $hasPermission], StatusCodeInterface::STATUS_OK);
+        return new PrivateJsonResponse(['hasPermission' => $hasPermission], StatusCodeInterface::STATUS_OK);
```

Comment:
@JonasVHG to clarify my previous comment, you can just use `JsonResponse` and not add the `cache-control` header in my opinion